pkg/proxy/headers/headers.go

```diff
@@ -162,7 +162,7 @@ func (l Lookup) Clone() Lookup {
 // If a key exists in both maps, the source value wins.
 // If the destination map is nil, the source map will not be merged
 func Merge(dst, src http.Header) {
-	if src == nil || len(src) == 0 || dst == nil {
+	if len(src) == 0 || dst == nil {
```

Comment:
Would this be more consistent:

```suggestion
	if len(src) == 0 || len(dst) {
```
Or have I missed something in the diff